### PR TITLE
PLANNER-865: Workbench: ScoreHolder support in GuidedDecisionTables causes error in server log

### PR DESCRIPTION
**Pull request by mcimbora**
Part of:
- https://github.com/kiegroup/drools/pull/1476
- https://github.com/kiegroup/drools-wb/pull/620
- https://github.com/kiegroup/optaplanner-wb/pull/222
- https://github.com/kiegroup/droolsjbpm-integration/pull/1193

### CODE REVIEW

**Comment by mcimbora** [comment]
@manstis @jomarko This PR solves the problem of drools compiler not being able to use persistence extensions for custom planner IActions. The actions already have enough information to marshal themselves, no need to use persistence extensions for IAction -> String conversion. Can you please take a look? Happy to answer any questions, thanks in advance.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/GuidedDecisionTableSourceService.java [hunk 1]
```diff
@@ -64,22 +64,17 @@
 
     private KieProjectService projectService;
 
-    private Collection<RuleModelIActionPersistenceExtension> persistenceExtensions = new ArrayList<>();
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/GuidedDecisionTableSourceService.java]: There remained unused imports:
- org.drools.workbench.models.commons.backend.rule.RuleModelIActionPersistenceExtension;
- javax.enterprise.inject.Instance;

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/indexing/GuidedDecisionTableFileIndexer.java [hunk 2]
```diff
@@ -43,16 +42,6 @@
     @Inject
     protected DataModelService dataModelService;
 
-    @Inject
-    protected Instance<RuleModelIActionPersistenceExtension> persistenceExtensionInstance;
-
-    private List<RuleModelIActionPersistenceExtension> persistenceExtensions = new ArrayList<>();
-
-    @PostConstruct
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/indexing/GuidedDecisionTableFileIndexer.java]: Also here some unused imports

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/test/java/org/drools/workbench/screens/guided/dtable/backend/server/GuidedDecisionTableSourceServiceTest.java [hunk 3]
```diff
@@ -99,15 +96,11 @@
 
     @Before
     public void setUp() throws Exception {
-        Instance<RuleModelIActionPersistenceExtension> persistenceExtensionInstance = mock(Instance.class);
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/test/java/org/drools/workbench/screens/guided/dtable/backend/server/GuidedDecisionTableSourceServiceTest.java]: Also here unused imports

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-backend/src/main/java/org/drools/workbench/screens/guided/rule/backend/server/GuidedRuleDRLSourceService.java [hunk 4]
```diff
@@ -42,13 +42,6 @@
     @Inject
     private GuidedRuleEditorService guidedRuleEditorService;
 
-    private Collection<RuleModelIActionPersistenceExtension> persistenceExtensions = new ArrayList<>();
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-backend/src/main/java/org/drools/workbench/screens/guided/rule/backend/server/GuidedRuleDRLSourceService.java]: Unused imports

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-backend/src/main/java/org/drools/workbench/screens/guided/rule/backend/server/GuidedRuleDSLRSourceService.java [hunk 5]
```diff
@@ -31,7 +31,6 @@
 import org.drools.compiler.lang.dsl.DSLTokenizedMappingFile;
 import org.drools.compiler.lang.dsl.DefaultExpander;
 import org.drools.workbench.models.commons.backend.rule.RuleModelDRLPersistenceImpl;
-import org.drools.workbench.models.commons.backend.rule.RuleModelIActionPersistenceExtension;
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-backend/src/main/java/org/drools/workbench/screens/guided/rule/backend/server/GuidedRuleDSLRSourceService.java]: Unused import: javax.enterprise.inject.Instance;

probably not changed by your changes, but would you mind remove it?

**Comment by mcimbora** [comment]
@jomarko @manstis All green everywhere. Thanks for the review.

**Pull request merged**